models/models.py

```diff
@@ -1,37 +1,42 @@
 import logging
 
-from sqlalchemy import MetaData, create_engine
-from sqlalchemy.ext.declarative import declarative_base
-from sqlalchemy.orm import Session
+from sqlalchemy import BigInteger, ForeignKey, MetaData, create_engine
+from sqlalchemy.orm import DeclarativeBase, MappedAsDataclass, Session, mapped_column
+from typing_extensions import Annotated
 
 from config import CONFIG
 
-convention = {
-    "ix": "ix_%(column_0_label)s",
-    "uq": "uq_%(table_name)s_%(column_0_name)s",
-    "ck": "ck_%(table_name)s_%(constraint_name)s",
-    "fk": "fk_%(table_name)s_%(column_0_name)s_%(referred_table_name)s",
-    "pk": "pk_%(table_name)s",
-}
+# this is bad, redo this
+engine = create_engine(CONFIG.DATABASE_CONNECTION, future=True)
 
-metadata = MetaData(naming_convention=convention)
-Base = declarative_base(metadata=metadata)
+from config import CONFIG
 
 engine = create_engine(CONFIG.DATABASE_CONNECTION)
 if CONFIG.SQL_LOGGING:
     logging.basicConfig()
     logging.getLogger("sqlalchemy.engine").setLevel(logging.WARNING)
-db_session = Session(bind=engine)
-
-
-def auto_str(cls):
-    def __repr__(self):
-        value = ", ".join(
-            "{}={}".format(*item)
-            for item in vars(self).items()
-            if item[0] != "_sa_instance_state"
-        )
-        return f"{type(self).__name__} {{{value}}}"
-
-    cls.__repr__ = __repr__
-    return cls
+db_session = Session(bind=engine, future=True)
+
+# some useful re-usable column types
+int_pk = Annotated[int, mapped_column(primary_key=True, init=False)]
+user_id = Annotated[int, mapped_column(ForeignKey("users.id"))]
+# discord 'snowflakes' are the really long IDs that you get as like channel or user IDs
+# this annotated column map type uses BigInteger to encode those and convert to python ints
+discord_snowflake = Annotated[int, mapped_column(BigInteger)]
+
```

Comment:
I believe the convention for custom types is to `PascalCase` them like class names. Even if it isn't convention, I generally prefer my custom types capitalized

Reply:
Just went to have a look, my memory serves correct for once. https://peps.python.org/pep-0008/#type-variable-names

Reply:
I did wonder about this, but I went with what the sqlalchemy docs used. I'll change it.